tests/s2_inference/test_clip_utils.py

```diff
@@ -41,3 +46,28 @@ def run():
                 return True
 
             run()
+
+    def test_load_image_from_path_http_error(self):
+        good_url = 'https://raw.githubusercontent.com/marqo-ai/marqo-api-tests/mainline/assets/ai_hippo_realistic.png'
+        # it should be fine normally
+        clip_utils.load_image_from_path(good_url, {})
+        #
+        normal_response = requests.get(good_url)
+        assert normal_response.status_code == 200
+
+        for status_code in itertools.chain(range(400, 452), range(500, 512)):
+            mock_get = mock.MagicMock()
+            bad_response = copy.deepcopy(normal_response)
+            bad_response.status_code = status_code
+            mock_get.return_value = bad_response
+
+            @mock.patch('requests.get', mock_get)
+            def run():
+                try:
+                    clip_utils.load_image_from_path(good_url, {})
+                    raise AssertionError
+                except PIL.UnidentifiedImageError:
+                    pass
```

Comment:
do you think we should return the error code from the HTTP request and add it to the error?

Reply:
The error code is propagated here: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/s2_inference/clip_utils.py#L110

I'll add this to the test

Reply:
Thanks! I was actually looking at the wrong function and didn't notice this error code propagation.

Reply:
added HTTP error code assertion in other test: https://github.com/marqo-ai/marqo/pull/402/files#diff-2bb102914165cddf1cfa00d25eb21234795d98c9b3c2f513ce4c14acae701a8eR70